src/Module/HCard.php

```diff
@@ -88,5 +77,20 @@ public static function rawContent(array $parameters = [])
 		foreach (['request', 'confirm', 'notify', 'poll'] as $dfrn) {
 			$page['htmlhead'] .= "<link rel=\"dfrn-{$dfrn}\" href=\"" . $baseUrl->get() . "/dfrn_{$dfrn}/{$nickname}\" />\r\n";
 		}
+
+		$block = (DI::config()->get('system', 'block_public') && !Session::isAuthenticated());
+
+		// check if blocked
+		if ($block) {
+			$keywords = $profile['pub_keywords'] ?? '';
+			$keywords = str_replace([',', ' ', ',,'], [' ', ',', ','], $keywords);
+			if (strlen($keywords)) {
+				$page['htmlhead'] .= '<meta name="keywords" content="' . $keywords . '" />' . "\r\n";
+			}
+		}
+
+		$page['aside'] = Profile::getVCardHtml($profile, $block, false);
```

Comment:
This functionality doesn't provide all the needed information for the HCard. There are many hidden fields there.

Reply:
That's literally all that `Profile::load` was doing for the `HCard` module, everything else was changing the active theme and setting the HTML title of the page. If there's something to change to this module, it's out of the scope of this PR.

Reply:
Ah, okay. I hadn't realized that `getVCardHtml` is the renamed `sidebar` function.